_main.cfg

```diff
@@ -67,6 +67,26 @@
     [lua]
         code = << wesnoth.dofile("~add-ons/Legend_of_the_Invincibles/lua/main.lua") >>
     [/lua]
+## EFFECT TO CHANGE LEVEL
+## Use on [effect]
+## apply_to=level
+## increase=number or $formula
+## set=number or $formula
+#ifver WESNOTH_VERSION < 1.17.15
+[lua]
+    code=<<
+        function wesnoth.effects.level(unit, cfg)
+            if cfg.set then
+                unit.level = cfg.set
+            elseif cfg.increase then
+                unit.level = unit.level + cfg.increase
+            else
+                wml.error("Invalid or missing key in [effect] apply_to=level")
+            end
+        end
+	>>
+[/lua]
+#endif
```

Comment:
Can you please put this in a separate file? I think this will be used much more once 1.18 goes out.

Reply:
Ok,  so another file on the lua folder

Reply:
Yeah.